File: examples/javaone-2026/src/main/java/org/example/koog/java/structs/AccountIssueSolution.java
package org.example.koog.java.structs;

import ai.koog.agents.core.tools.annotations.LLMDescription;
import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

@LLMDescription("Summary about what has been done to resolve the account issue")
public record AccountIssueSolution(
        @JsonProperty("accountNumber")
        @LLMDescription("Account number that was affected")
        String accountNumber,

        @JsonProperty("actionsTaken")
        @LLMDescription("Brief summary of the actions taken to resolve the issue")
        String actionsTaken
) {
    @JsonCreator
    public AccountIssueSolution {}

    public static AccountIssueSolution empty() {
        return new AccountIssueSolution("account-123", "Nothing changed");
    }
}

File: examples/spring-io-2026/kotlin/src/main/kotlin/org/example/koog/spring/kotlin/structs/AccountIssueSolution.kt
package org.example.koog.spring.kotlin.structs

import ai.koog.agents.core.tools.annotations.LLMDescription
import com.fasterxml.jackson.annotation.JsonCreator
import kotlinx.serialization.Serializable

@LLMDescription("Summary about what has been done to resolve the account issue")
@Serializable
data class AccountIssueSolution @JsonCreator constructor(
    @property:LLMDescription("Account number that was affected")
    val accountNumber: String,
    @property:LLMDescription("Brief summary of the actions taken to resolve the issue")
    val actionsTaken: String
)
